File: frontend/src/types/announcement.ts
import type { User } from './user';

export type AnnouncementCategory = 'notice' | 'event' | 'urgent';

export interface Announcement {
  id: number;
  title: string;
  content: string;
  category: AnnouncementCategory;
  publisherId: number;
  publishAt: string;
  top: boolean;
  readCount: number;
  publisher?: User;
}

File: backend/src/main/java/com/smartestate/entity/Announcement.java
package com.smartestate.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("announcements")
public class Announcement {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String content;
    private String category;
    private Long publisherId;
    private LocalDateTime publishAt;
    private Boolean top;
    private Integer readCount;

    @TableField(exist = false)
    private User publisher;
}
